PROTO_tests/tests/setops_test.py

```diff
@@ -0,0 +1,783 @@
+        lr = [x in lb for x in la]
+        assert ak_result.to_list() == lr
+
+        stringsOne = ak.array(["String {}".format(i % 3) for i in range(10)])
+        stringsTwo = ak.array(["String {}".format(i % 2) for i in range(10)])
+        assert [(x % 3) < 2 for x in range(10)] == ak.in1d(stringsOne, stringsTwo).to_list()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_in1d_multiarray_categorical(self, size):
+        # large general test
+        a = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+        b = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+
+        c = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+        d = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+
+        l1 = [a, b]
+        l2 = [c, d]
+
+        ak_result = ak.in1d(l1, l2)
+
+        la = [(x, y) for x, y in zip(a.to_list(), b.to_list())]
+        lb = [(x, y) for x, y in zip(c.to_list(), d.to_list())]
+        lr = [x in lb for x in la]
+        assert ak_result.to_list() == lr
+
+        stringsOne = ak.Categorical(ak.array(["String {}".format(i % 3) for i in range(10)]))
+        stringsTwo = ak.Categorical(ak.array(["String {}".format(i % 2) for i in range(10)]))
+        assert [(x % 3) < 2 for x in range(10)] == ak.in1d(stringsOne, stringsTwo).to_list()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    def test_intersect1d_multiarray_numeric_types(self, size, dtype):
+        a, b = self.make_np_arrays(size, dtype)
+        c, d = self.make_np_arrays(size, dtype)
+
+        l1 = [ak.array(a), ak.array(c)]
+        l2 = [ak.array(b), ak.array(d)]
+
+        ak_result = ak.intersect1d(l1, l2)
+
+        la = set([(x, y) for x, y in zip(a, c)])
+        lb = set([(x, y) for x, y in zip(b, d)])
+        lr = list(sorted(la.intersection(lb)))
+        ak_result = [x.to_list() for x in ak_result]
+        ak_result = list(zip(*ak_result))
+        assert ak_result == lr
+
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    def test_intersect1d_multiarray_numeric_small(self, dtype):
+        a, b, c, d = self.make_np_arrays_multi_small(dtype)
+
+        l1 = [ak.array(a), ak.array(b)]
+        l2 = [ak.array(c), ak.array(d)]
+        ak_result = ak.intersect1d(l1, l2)
+
+        la = set([(x, y) for x, y in zip(a, b)])
+        lb = set([(x, y) for x, y in zip(c, d)])
+        lr = list(sorted(la.intersection(lb)))
+
+        ak_result = [x.to_list() for x in ak_result]
+        # sorting applied for bigint case. Numbers are right, but not ordering properly
+        ak_result = sorted(list(zip(*ak_result)))
+        assert ak_result == lr
+
+    @pytest.mark.parametrize("dtype1", INTEGRAL_TYPES)
+    @pytest.mark.parametrize("dtype2", INTEGRAL_TYPES)
+    def test_intersect1d_multiarray_cross_type(self, dtype1, dtype2):
+        a, c = self.make_np_arrays_cross_type(dtype1)
+        b, d = self.make_np_arrays_cross_type(dtype2)
```

Comment:
same comment about potentially being able to remove `test_intersect1d_multiarray_numeric_small` if we modify `make_np_arrays_cross_type` to accept two dtypes